packages/@jsii/check-node/src/index.ts

```diff
@@ -22,12 +22,13 @@ export function checkNode(envPrefix = 'JSII'): void {
 
   if (nodeRelease?.endOfLife) {
     const silenceVariable = `${envPrefix}_SILENCE_WARNING_END_OF_LIFE_NODE_VERSION`;
-    const acknowledgeNodeEol =
-      'Node14 is now end of life (EOL) as of April 30, 2023. Support of EOL runtimes are only guaranteed for 30 days after EOL. By silencing this warning you acknowledge that you are using an EOL version of Node and will upgrade to a supported version as soon as possible.';
+    const silencedVersions = (process.env[silenceVariable] ?? '')
+      .split(',')
+      .map((v) => v.trim());
```

Comment:
The previous implementation was very verbose and used a "nanny" approach. In reality we need this check only to silence warnings in tests. It did also not account for different versions.

Re-implemented this to require ack of every specific node version.